RELEASE_NOTES.rst

```diff
@@ -1,3 +1,133 @@
+
+We have added a release candidate system to our release process to allow users
+to easily test changes in the latest pre-release version of coala.
+Instructions for obtaining the rc package will be posted for each minor/major
+release to the coala gitter channel, blog, and twitter. We encourage all users
+interested in checking out new features or helping to test these pre-releases
+on their projects. If you do find a regression in a release candidate, please
+report it to the issue tracker with the word ``Regression`` somewhere in the
+title.
+
+**Breaking Changes**
+
+- coala now exits with a non zero exit code when results are found.
+  If you relied on coala in scripts or other automated systems please
+  modify them to work with the new behavior. You can find all exit codes
+  `here <http://docs.coala.io/en/latest/Users/Exit_Codes.html>`__.
+  `PR #3905 <https://github.com/coala/coala/pull/3905>`_
+
+**General**
+
+- Section Inheritance had been added to the coafile parsing. With this
+  change we are deprecating implicit ``default`` section inheritance.
+  Instead, you can now have definite section inheritance by using the
+  ``.`` in the section title. Append operators are now supported allowing
+  addition of values to inherited section. See the
+  ``Explicit Setting Inheritance`` section of the
+  `docs <https://docs.coala.io/en/latest/Users/coafile.html#explicit-setting-inheritance>`__
+  for a syntax example and detailed instructions
+- The section name ``cli`` is reserved for cli arguments and should not be
+  used in the coala configuration file (``coafile``). If you are currently
+  using it in a configuration file, coala will warn you.
+- coala now outputs absolute file paths when a bear fails. This should
+  make debugging bear failures easier when using custom bears.
+
+**Usability**
+
+- A new placeholder, ``source_lines``,  has been added to the coala
+  ``--format`` option. Use it to output the affected lines in the source file
+  when a result occurs.
+- Bears are displayed alphabetically when listed using ``--show-bears``.
+- coala outputs a warning if a setting is overridden in the same section
+  since this is probably not what the user intended.
+- ``--format`` is handled correctly when ``--ci`` is specified, outputting
+  formatted output without asking questions.
+  `Issue #3999 <https://github.com/coala/coala/issues/3999>`_
+- coala will report the version of the dependency required when the version
+  check fails.
+  `Issue #3982 <https://github.com/coala/coala/issues/3982>`_
+
+**Bear Developers**
+
+- ``@linter`` decorator works with global bears. See our bear writing
+  `docs <http://api.coala.io/en/latest/Developers/Writing_Linter_Bears.html#global-bears>`__
+  for more information on how to write a global bear with the ``@linter``
+  decorator.
+- ``Bear.DownloadCachedFile`` raises exceptions when the download fails.
+  Please handle this exception if the file download has a good chance
```

Comment:
there are a few. These exceptions are actually raw requests library exceptions. WE just don't handle them yet